### ISSUE TITLE
no-unused-vars  / no-undef with JSX and 0.16.x

### ISSUE THREAD

**Issue by MoOx** [eslint/eslint#2004]
eslint 0.16.2

`src/widget.js`

```jsx
"use strict";

var React = require("react")
var HelloWorld = require("op/HelloWorld")

React.render(
  <div className="op-Widget">
    <HelloWorld />
  </div>,
  document.getElementById("op-Widget")
)
```

`.eslintrc`

```yaml
ecmaFeatures:
  modules: true
  jsx: true

env:
  es6: true
  browser: true
  node: true

globals:
  __DEV__: true
  __PROD__: true

rules:
  semi: 0
  quotes: [2, "double"]
  indent: [2, 2]
  comma-dangle: [2, "always"]
  comma-style: [2, "last"]
  valid-jsdoc: 1
```

Gives

```console
$ eslint src/widget.js

src/widget.js
   8:4  error  HelloWorld is defined but never used     no-unused-vars
  12:5  error  'HelloWorld' is not defined              no-undef

✖ 2 problems (2 errors, 0 warnings)
```

I do have the issue with 0.16.0 and 0.16.1 too
I've tried to disable modules & es6 but I am getting the same behavior.

**Comment by lo1tuma** [comment]
Duplicate of #1997, #1982 and #1976

**Comment by MoOx** [comment]
Ok sorry for this.

**Comment by nzakas** [comment]
See #1911 as well.
